src/utils/Assets.ts

```diff
@@ -0,0 +1,191 @@
+import { SHA256 } from 'crypto-js'
+import { ethers, Signer } from 'ethers'
+import {
+  Aquarius,
+  DatatokenCreateParams,
+  Nft,
+  NftCreateData,
+  NftFactory,
+  ProviderInstance,
+  ZERO_ADDRESS,
+  // approveWei,
+  // ProviderComputeInitialize,
+  // ConsumeMarketFee,
+  // Datatoken,
+  // Config,
+  // DDO,
+  // ProviderFees,
+  getEventFromTx,
+  DispenserCreationParams,
+  FreCreationParams,
+  ConfigHelper
+} from '../../src'
+import { hexlify } from 'ethers/lib/utils'
+import { createHash } from 'crypto'
+
+const VALID_TEMPLATE_INDEXES = [1, 2, 4]
+/**
+ *
+ * @param name asset name
+ * @param symbol asse symbol
+ * @param owner owner address
+ * @param assetUrl asset url
+ * @param templateIndex 1,2 or 4
+ * @param ddo ddo
+ * @param encryptDDO encrypt or not?
+ * @param providerUrl the provider URL
+ * @param providerFeeToken the provider fee token
+ * @param nftContractAddress the nft contract address
+ * @param aquariusInstance aquarius, could be node instance url
+ * @param dispenserAddress dispenser address
+ * @param fixedRateAddress fixed rate exchange address
+ * @param baseTokenAddress base token address (ocean)
+ * @returns ddo id as string
+ */
+export async function createAsset(
+  name: string,
+  symbol: string,
+  owner: Signer,
+  assetUrl: any,
+  templateIndex: number,
+  ddo: any,
+  encryptDDO: boolean = true, // default is true
+  providerUrl: string,
+  providerFeeToken: string,
+  nftContractAddress: string, // addresses.ERC721Factory,
+  aquariusInstance: Aquarius,
+  filesObject?: any
+  // fixed rate
+  // dispenserAddress?: string,
+  // fixedRateAddress?: string,
+  // baseTokenAddress?: string // ocean token?
+): Promise<string> {
+  if (!VALID_TEMPLATE_INDEXES.includes(templateIndex)) {
+    throw new Error('Invalid template index: ' + templateIndex)
+  }
+
+  const chainID = (await owner.provider.getNetwork()).chainId
+
+  const config = new ConfigHelper().getConfig(parseInt(String(chainID)))
+
+  const nft = new Nft(owner, chainID)
+
+  const nftFactory = new NftFactory(nftContractAddress, owner)
+
+  // get nft owner
+  const account = await owner.getAddress()
+
+  // from hex to number format
+  ddo.chainId = parseInt(chainID.toString(10))
+  const nftParamsAsset: NftCreateData = {
+    name,
+    symbol,
+    templateIndex,
+    tokenURI: 'aaa',
+    transferable: true,
+    owner: account
+  }
+  const datatokenParams: DatatokenCreateParams = {
+    templateIndex,
```

Comment:
templateIndex should be calculated.

Don't go to "templateIndex == templateId", this will fail.

For instance, on sapphire we have the following config now:  (https://github.com/oceanprotocol/contracts/blob/main/addresses/address.json#L261C3-L282C5)

```
oasis_saphire_testnet": {
    "chainId": 23295,
    "Ocean": "0x973e69303259B0c2543a38665122b773D28405fB",
    "OPFCommunityFeeCollector": "0xe8c6Dc39602031A152440311e364818ba25C2Bc1",
    "startBlock": 2565487,
    "Router": "0x5FBC2A29f7C8e4533dECD14Dc1c561A8eF0d4e31",
    "FixedPrice": "0x0846aec7aADD547131484Bf7fE548589b7F057fe",
    "ERC20Template": {
      "1": "0x519d462294958A935D34351b3ad9F98cCAB0b7f8",
      "2": "0x2bCa501a6f9FB8a96f1b7B09D4f74A32ACAa7584",
      "3": "0xf00c16665b79F23c04e5c8652Fe9853046b987c8",
      "4": "0x89b752C3e272feBA17B363EaaBFe0d653cF75bE7",
      "5": "0xb4914dfeC7f70C708673620eDF6d8839F81E7612"
    },
    "ERC721Template": {
      "1": "0x9Ab15cc945b2B58A51384539211dE8a88a1355E8"
    },
    "Dispenser": "0xbde8224d15885c2E46Be665b9392E619cD142de8",
    "ERC721Factory": "0x315ef329D214cde5cE7C206104aC35B1fe66bb8c",
    "PredictoorHelper": "0x2e4cb19faD077768fDcB04Bc0549AB76BE8eA63F",
    "AccessListFactory": "0xdA412123a3Cc0D921bcf7246f8742e3b7c3E7b94"
  },
  ```
  
  Guess what?  templateIndex 4 is disabled..
  
  ```bash
  Using the following as start point:
{
  chainId: 23295,
  Ocean: '0x973e69303259B0c2543a38665122b773D28405fB',
  OPFCommunityFeeCollector: '0xe8c6Dc39602031A152440311e364818ba25C2Bc1',
  startBlock: 2565487,
  Router: '0x5FBC2A29f7C8e4533dECD14Dc1c561A8eF0d4e31',
  FixedPrice: '0x0846aec7aADD547131484Bf7fE548589b7F057fe',
  ERC20Template: {
    '1': '0x519d462294958A935D34351b3ad9F98cCAB0b7f8',
    '2': '0x2bCa501a6f9FB8a96f1b7B09D4f74A32ACAa7584',
    '3': '0xf00c16665b79F23c04e5c8652Fe9853046b987c8',
    '4': '0x89b752C3e272feBA17B363EaaBFe0d653cF75bE7',
    '5': '0xb4914dfeC7f70C708673620eDF6d8839F81E7612'
  },
  ERC721Template: { '1': '0x9Ab15cc945b2B58A51384539211dE8a88a1355E8' },
  Dispenser: '0xbde8224d15885c2E46Be665b9392E619cD142de8',
  ERC721Factory: '0x315ef329D214cde5cE7C206104aC35B1fe66bb8c',
  PredictoorHelper: '0x2e4cb19faD077768fDcB04Bc0549AB76BE8eA63F',
  AccessListFactory: '0xdA412123a3Cc0D921bcf7246f8742e3b7c3E7b94'
}
nftCount
BigNumber { value: "1" }
Router owner: 0xC7EC1970B09224B317c52d92f37F5e1E4fF6B687
currentTokenCount
BigNumber { value: "5" }


------------
templateIndex:1
[
  '0x519d462294958A935D34351b3ad9F98cCAB0b7f8',
  true,
  templateAddress: '0x519d462294958A935D34351b3ad9F98cCAB0b7f8',
  isActive: true
]
templateId: 1


------------
templateIndex:2
[
  '0x2bCa501a6f9FB8a96f1b7B09D4f74A32ACAa7584',
  true,
  templateAddress: '0x2bCa501a6f9FB8a96f1b7B09D4f74A32ACAa7584',
  isActive: true
]
templateId: 2


------------
templateIndex:3
[
  '0xf00c16665b79F23c04e5c8652Fe9853046b987c8',
  true,
  templateAddress: '0xf00c16665b79F23c04e5c8652Fe9853046b987c8',
  isActive: true
]
templateId: 3


------------
templateIndex:4
[
  '0x89b752C3e272feBA17B363EaaBFe0d653cF75bE7',
  false,
  templateAddress: '0x89b752C3e272feBA17B363EaaBFe0d653cF75bE7',
  isActive: false
]
templateId: 4


------------
templateIndex:5
[
  '0xb4914dfeC7f70C708673620eDF6d8839F81E7612',
  true,
  templateAddress: '0xb4914dfeC7f70C708673620eDF6d8839F81E7612',
  isActive: true
]
templateId: 4
```


so, the proper way of determining templateIndex is :

```js

    const currentTokenCount = await factoryERC721.connect(owner).getCurrentTemplateCount(options);
    for(i=1;i<=currentTokenCount;i++){
        const tokenTemplate = await factoryERC721.connect(owner).getTokenTemplate(i,options);
        console.log("\n\n------------\ntemplateIndex:"+i)
        console.log(tokenTemplate)
        const erc20Template = await ethers.getContractAt("ERC20Template", tokenTemplate["templateAddress"]);
        const id=await erc20Template.connect(owner).getId()
        console.log("templateId: "+id)
        if(tokenTemplate.isActive && id==SEARCHED_TEMPLATE_ID || tokenTemplate["templateAddress"]==SEARCHED_TEMPLATE_ADDRESS){
                               return i
        }
    }
    return null     
    ```